adapters/http-vertx/src/main/java/org/eclipse/hono/adapter/http/impl/VertxBasedHttpProtocolAdapter.java

```diff
@@ -110,22 +113,23 @@ protected void addRoutes(final Router router) {
                     Optional.ofNullable(usernamePasswordAuthProvider).orElse(
                             new UsernamePasswordAuthProvider(getCredentialsClientFactory(), getConfig(), tracer)),
                     getConfig().getRealm(), tracer));
-            addTelemetryApiRoutes(router, authHandler);
-            addEventApiRoutes(router, authHandler);
-            addCommandResponseRoutes(CommandConstants.COMMAND_ENDPOINT, router, authHandler);
-            addCommandResponseRoutes(CommandConstants.COMMAND_LEGACY_ENDPOINT, router, authHandler);
+            addTelemetryApiRoutes(router, tenantTraceSamplingHandler, authHandler);
```

Comment:
wouldn't it be easier to just add it as a handler that is used for all requests? I guess the overhead is minimal as the handler never fails but just continues with the next handler if no Basic auth header is found in the request, right?

Reply:
Yes, I've amended the commit to use only one generic route handler in the base class.
I've also refactored the new `TenantTraceSamplingHandler` class so that logic can be reused in other protocol adapters, to be done in follow-up commits.